### PR TITLE
ESG-3737 SFCC|Create COD payment method

### PR DESCRIPTION
**Pull request by rikai-vietle**
Screenshot from 2021-10-05 09-30-27

Screenshot from 2021-10-05 09-30-32

### CODE REVIEW

File: cartridges/app_custom_subscstore/cartridge/models/payment.js [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+            results.type = paymentInstrument.creditCardType;
+            results.maskedCreditCardNumber = paymentInstrument.maskedCreditCardNumber;
+            results.expirationMonth = paymentInstrument.creditCardExpirationMonth;
+        } else if (paymentInstrument.paymentMethod === 'GIFT_CERTIFICATE') {
+            results.giftCertificateCode = paymentInstrument.giftCertificateCode;
+            results.maskedGiftCertificateCode = paymentInstrument.maskedGiftCertificateCode;
+        } else if (paymentInstrument.paymentMethod === 'COD') {
+            PaymentMgr = require('dw/order/PaymentMgr');
+            results.paymentMethodName = PaymentMgr.getPaymentMethod(paymentInstrument.paymentMethod).name;
+        }
```
Comment (rikai-vietle) [inline cartridges/app_custom_subscstore/cartridge/models/payment.js:64]: Add process paymentMethod === 'COD')

File: cartridges/app_custom_subscstore/cartridge/templates/default/checkout/billing/paymentOptions.isml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+<div class="form-nav billing-nav payment-information"
+     data-payment-method-id="COD"
```
Comment (rikai-vietle) [inline cartridges/app_custom_subscstore/cartridge/templates/default/checkout/billing/paymentOptions.isml:2]: Change default active tab CREDIT CARD to COD

File: cartridges/app_custom_subscstore/cartridge/templates/default/checkout/billing/paymentOptions/creditCardContent.isml [hunk 3]
```diff
@@ -0,0 +1,34 @@
+<!--- applicable credit cards--->
+<div class="tab-pane credit-card-content" id="credit-card-content" role="tabpanel">
```
Comment (rikai-vietle) [inline cartridges/app_custom_subscstore/cartridge/templates/default/checkout/billing/paymentOptions/creditCardContent.isml:2]: Remove default active tab CREDIT CARD

File: cartridges/app_custom_subscstore/cartridge/templates/default/checkout/billing/paymentOptions/creditCardTab.isml [hunk 4]
```diff
@@ -0,0 +1,10 @@
+<li class="nav-item" data-method-id="${paymentOption.ID}">
+    <a class="nav-link credit-card-tab" data-toggle="tab" href="#credit-card-content" role="tab">
```
Comment (rikai-vietle) [inline cartridges/app_custom_subscstore/cartridge/templates/default/checkout/billing/paymentOptions/creditCardTab.isml:2]: Remove default active tab CREDIT CARD

File: cartridges/app_custom_subscstore/cartridge/models/account.js [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+        var result = {
+            creditCardHolder: paymentInstrument.creditCardHolder,
+            maskedCreditCardNumber: paymentInstrument.maskedCreditCardNumber,
+            creditCardType: paymentInstrument.creditCardType,
+            creditCardExpirationMonth: paymentInstrument.creditCardExpirationMonth,
+            creditCardExpirationYear: paymentInstrument.creditCardExpirationYear,
+            UUID: paymentInstrument.UUID
+        };
+
+        if (paymentInstrument.creditCardType) {
```
Comment (rikai-vietle) [inline cartridges/app_custom_subscstore/cartridge/models/account.js:97]: Check creditCardType != null if payment is COD

File: cartridges/app_custom_subscstore/cartridge/client/default/js/checkout/billing.js [hunk 6]
```diff
@@ -0,0 +1,333 @@
...
+    var $paymentSummary = $('.payment-details');
+    var htmlToAppend = '';
+
+    if (order.billing.payment && order.billing.payment.selectedPaymentInstruments
+        && order.billing.payment.selectedPaymentInstruments.length > 0) {
+        if (order.billing.payment.selectedPaymentInstruments[0].paymentMethod === 'COD') {
+            htmlToAppend += '<span>'
+                + order.billing.payment.selectedPaymentInstruments[0].paymentMethodName
+                + '</span>';
+        } else {
```
Comment (rikai-vietle) [inline cartridges/app_custom_subscstore/cartridge/client/default/js/checkout/billing.js:193]: Show payment method COD at placeOrder

File: cartridges/app_custom_subscstore/cartridge/controllers/CheckoutServices.js [hunk 7]
```diff
@@ -1,225 +0,0 @@
-'use strict';
```
Comment (rikai-anhphan) [inline cartridges/app_custom_subscstore/cartridge/controllers/CheckoutServices.js:1]: sao lại xóa file này a nhỉ

Comment (rikai-vietle) [reply cartridges/app_custom_subscstore/cartridge/controllers/CheckoutServices.js]: Anh revert rồi nha

File: cartridges/app_custom_subscstore/cartridge/api/subscstore/build_params/buildCreateOrder.js [hunk 8]
```diff
@@ -6,16 +8,102 @@
...
+    };
+
+    var customPaymentMethod = PaymentMgr.getPaymentMethod(
+        order.paymentInstrument.paymentMethod).getCustom();
+    var sSpaymentMethodId = customPaymentMethod.SSpaymentMethodId;
+    var sSpaymentMethodShopId = customPaymentMethod.SSpaymentMethodShopId;
+    body.settlement = {
+        payment_method_id: sSpaymentMethodId,
+        payment_method_shop_id: sSpaymentMethodShopId
+    };
```
Comment (rikai-vietle) [inline cartridges/app_custom_subscstore/cartridge/api/subscstore/build_params/buildCreateOrder.js:49]: Fix get payment method shop id

**Review by tqhung04 — APPROVED** [review]
LGTM!

**Pull request merged**